rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -2626,6 +2639,32 @@ namespace rsx
 		return fifo_ctrl->last_cmd();
 	}
 
+	std::string thread::dump_regs() const
+	{
+		std::string result;
+
+		for (u32 i = 0; i < 1 << 14; i++)
+		{
+			switch (i)
+			{
+			case NV4097_NO_OPERATION:
+			case NV4097_INVALIDATE_L2:
+			case NV4097_INVALIDATE_VERTEX_FILE:
+			case NV4097_INVALIDATE_VERTEX_CACHE_FILE:
+			case NV4097_INVALIDATE_ZCULL:
+				continue;
+			default: break;
+			}
+
+			if (auto func = get_raw_printing_function(i))
+			{
+				fmt::append(result, "[%04x] %s\n", i, func(method_registers.registers[i]));
+			}
```

Comment:
If there is a chance of the method returning no function, we need a way to identify that from within the debugger.

Reply:
like say call rsx::get_methid_name for it and print that instead?

Reply:
yes.

Reply:
Done.

Reply:
My point was more about code integrity, since the code has been tightened up, add an assert here to avoid someone breaking it. We don't need the if() here, just an ensure, viz:
```
auto func = rsx::get_pretty_printing_function(i);
ensure(func);
//do something with func
```
Alternatively:
```
if (func = ..)
else fmt::throw...
```
Since I can see get_pretty_printing_function is not expected to fail, I propose the first version that just asserts.

Reply:
Fixed.